apps/projects/app/components/Content/Filters.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component, Fragment } from 'react'
+import PropTypes from 'prop-types'
+
+import FilterTile from './FilterTile'
+import { prepareFilters } from '../Shared/FilterBar'
+
+export default class Filters extends Component {
+  static propTypes = {
+    filters: PropTypes.shape({
+      projects: PropTypes.object.isRequired,
+      labels: PropTypes.object.isRequired,
+      milestones: PropTypes.object.isRequired,
+      deadlines: PropTypes.object.isRequired,
+      experiences: PropTypes.object.isRequired,
+      statuses: PropTypes.object.isRequired,
+    }),
+    issues: PropTypes.array,
+    bountyIssues: PropTypes.array,
+    disableFilter: PropTypes.func.isRequired,
+  }
+
+  static defaultProps = {
+    filters: {
+      projects: {},
+      labels: {},
+      milestones: {},
+      deadlines: {},
+      experiences: {},
+      statuses: {}
+    },
+    issues: [],
+    bountyIssues: [],
+  }
+
+  generateFilterNamesAndPaths = (filterInformation, type, textFieldToUse) => {
+    const { filters } = this.props
+    const appliedFilters = {}
+    Object.keys(filterInformation[type]).map(id => {
+      const filterApplied = id in filters[type]
+      const filterName = filterInformation[type][id][textFieldToUse]
+      if (filterApplied) {
+        appliedFilters[filterName] = [ type, id ]
+      }
+    })
+    return appliedFilters
+  }
+
+  /*
+    creates one object that looks like this:
+    {
+      filter1Name: [path, to, filter1, on, parent],
+      filter2Name: [path, to, filter2, on, parent],
+    }
+
+    to make it easier to deselect filters from this view without multiple state objects
+  */
+  calculateFilters = () => {
+    const { issues, bountyIssues } = this.props
+    const filterInformation = prepareFilters(issues, bountyIssues)
```

Comment:
This `prepareFilters` functionality I stripped out of the `FilterBar` component and reused it here - it provides the information about filters that the UI needs to display stuff. This is working really well, but I would still love to abstract all of this logic out and store it all in one state variable on the parent `Issues.js` or use the render props pattern. Feeling like that might be overkill for the moment. Happy to do more abstraction and cleanup around this tho. Just lmk